docs/source/API/core/atomics/atomic_fetch_op.rst

```diff
@@ -0,0 +1,104 @@
+``atomic_fetch_[op]``
+=====================
+
+.. role:: cppkokkos(code)
+   :language: cppkokkos
+
+Header File: <Kokkos_Core.hpp>
+
+Usage
+-----
+
+.. code-block:: cpp
+
+   old_value =  atomic_fetch_[op](ptr_to_value, update_value);
+
+Atomically updates the variable at the address given by ``ptr_to_value`` with ``update_value``
+according to the relevant operation, and returns the previous value found at that address.
+
+Description
+-----------
+
+.. cppkokkos:function:: template<class T> T atomic_fetch_add(T* const ptr_to_value, const T value);
+
+   Atomically executes ``tmp = *ptr_to_value; *ptr_to_value += value; return tmp;``.
+
+   :param ptr_to_value: address of the value to be updated
+   :param value: value to be added
+
+.. cppkokkos:function:: template<class T> T atomic_fetch_and(T* const ptr_to_value, const T value);
```

Comment:
Typo?  Should `atomic_fetch_and` (sic) be `atomic_fetch_add`?

Reply:
no, this is doing ``&=`` not ``+=``